src/main/java/io/cryostat/discovery/ContainerDiscovery.java

```diff
@@ -150,19 +187,18 @@ public abstract class ContainerDiscovery {
 
     protected long timerId;
 
-    protected final CopyOnWriteArrayList<ContainerSpec> containers = new CopyOnWriteArrayList<>();
-
+    // Priority is set higher than default 0 such that onStart is called first before onAfterStart
+    // This ensures realm node is persisted before querying for containers
     @Transactional
-    void onStart(@Observes StartupEvent evt) {
+    void onStart(@Observes @Priority(1) StartupEvent evt) {
```

Comment:
Ah for this, I was just hoping to ensure realm node is persisted before any discovery operations. I guess if the request to podman/docker api takes enough time, it should be fine to combine?

Reply:
Or sorry, rather than at the end, this one would go at the beginning. Like this:

```java
    @Transactional
    void onStart(@Observes StartupEvent evt) {
        if (!enabled()) {
            return;
        }

        if (!available()) {
            logger.errorv(
                    "{0} enabled but socket {1} is not accessible!",
                    getClass().getName(), Path.of(getSocket().path()));
            return;
        }

        DiscoveryNode universe = DiscoveryNode.getUniverse();
        if (DiscoveryNode.getRealm(getRealm()).isEmpty()) {
            DiscoveryPlugin plugin = new DiscoveryPlugin();
            DiscoveryNode node = DiscoveryNode.environment(getRealm(), BaseNodeType.REALM);
            plugin.realm = node;
            plugin.builtin = true;
            universe.children.add(node);
            node.parent = universe;
            plugin.persist();
            universe.persist();
        }

        logger.infov("Starting {0} client", getRealm());

        queryContainers();
        this.timerId = vertx.setPeriodic(pollPeriod.toMillis(), unused -> queryContainers());
    }
```